src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GetSubmissionTest.java

```diff
@@ -0,0 +1,132 @@
+import org.springframework.http.ResponseEntity;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DisplayName("SubmissionApiDelegateImpl test suite")
+public class GetSubmissionTest {
+
+    private static final String SUCCESSURL = "http://success";
+    private static final String CANCELURL = "http://cancel";
+    private static final String ERRORURL = "http://error";
+    private static final UUID CASE_1 = UUID.fromString("77da92db-0791-491e-8c58-1a969e67d2fe");
+    private static final UUID CASE_2 = UUID.fromString("77da92db-0791-491e-8c58-1a969e67d2fa");
+    private static final UUID CASE_3 = UUID.fromString("77da92db-0791-491e-8c58-1a969e67d2fb");
+    private static final String EMAIL = "email";
+    private static final String FIRST_NAME = "firstName";
+    private static final String LAST_NAME = "lastName";
+    private static final String MIDDLE_NAME = "middleName";
+
+    private SubmissionApiDelegateImpl sut;
+
+    @Mock
+    private SubmissionService submissionServiceMock;
+
+    @Mock
+    private GenerateUrlResponseMapper generateUrlResponseMapperMock;
+
+    @Mock
+    private SubmissionStore submissionStoreMock;
+
+    @BeforeAll
+    public void setUp() {
+
+        MockitoAnnotations.initMocks(this);
+
+        NavigationProperties navigationProperties = new NavigationProperties();
+        navigationProperties.setBaseUrl("http://localhost");
+
+        Submission submissionWithCsoAccount = Submission
+                .builder()
+                .accountDetails(
+                        AccountDetails
+                                .builder()
+                                .accountId(BigDecimal.TEN)
+                                .firstName(FIRST_NAME + CASE_2)
+                                .lastName(LAST_NAME + CASE_2)
+                                .middleName(MIDDLE_NAME + CASE_2)
+                                .fileRolePresent(true)
+                                .email(EMAIL + CASE_2).create())
+                .navigation(TestHelpers.createNavigation(SUCCESSURL, CANCELURL, ERRORURL))
+                .create();
+
+        Mockito.when(submissionStoreMock.getByKey(CASE_2)).thenReturn(Optional.of(submissionWithCsoAccount));
+
+        Submission submissionWithoutCsoAccount = Submission
+                .builder()
+                .navigation(TestHelpers.createNavigation(SUCCESSURL, CANCELURL, ERRORURL))
+                .create();
+
+        Mockito.when(submissionStoreMock.getByKey(CASE_3)).thenReturn(Optional.of(submissionWithoutCsoAccount));
+
+        sut = new SubmissionApiDelegateImpl(submissionServiceMock, generateUrlResponseMapperMock, navigationProperties, submissionStoreMock);
+
+    }
+
+    @Test
+    @DisplayName("With null redis storage response return NotFound")
```

Comment:
non-blocking: display names with http status codes in the previous file are a bit more descriptive than this one.